### PR TITLE
Fix means of logps and rewards

### PR DESCRIPTION
**Pull request by vaibhavjindal**
## Summary

The means of chosen/rejected logps and chosen/rejected rewards are being calculated incorrectly. This PR fixes this by taking care of the fact that the shapes of different tensors are as follows:
logps: (batch_size,)
logits: (batch_size, seq_len, vocab_size)
rewards: (batch_size, )

## Testing Done

- Hardware Type: <BLANK>
- [x] run `make test` to ensure correctness
- [x] run `make checkstyle` to ensure code style
- [ ] run `make test-convergence` to ensure convergence

### CODE REVIEW

**Review by shivam15s — APPROVED** [review]
lgtm
